composer/loggers/mlflow_logger.py

```diff
@@ -39,67 +44,100 @@ def __init__(
         run_name: Optional[str] = None,
         tracking_uri: Optional[Union[str, pathlib.Path]] = None,
         rank_zero_only: bool = True,
+        flush_interval: int = 10,
     ) -> None:
         try:
             import mlflow
+            from mlflow import MlflowClient
+            from mlflow.utils.autologging_utils import MlflowAutologgingQueueingClient
         except ImportError as e:
             raise MissingConditionalImportError(extra_deps_group='mlflow',
                                                 conda_package='mlflow',
                                                 conda_channel='conda-forge') from e
-        del mlflow
         self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
 
         self.run_name = run_name
         self.experiment_name = experiment_name
         self._rank_zero_only = rank_zero_only
-        self.tracking_uri = tracking_uri
+        self.tracking_uri = str(tracking_uri or mlflow.get_tracking_uri())
```

Comment:
Should this be gated behind the enable check 3 lines down as well? Not sure if this pings server or something 🤷